crates/kubelet/src/pod/queue.rs

```diff
@@ -19,79 +30,166 @@ use crate::provider::Provider;
 /// modify will be handled, which is ok given that each event contains the whole pod object
 pub(crate) struct Queue<P> {
     provider: Arc<P>,
-    handlers: HashMap<String, Worker>,
-    error_sender: Sender<(KubePod, anyhow::Error)>,
+    handlers: HashMap<String, tokio::sync::mpsc::Sender<WatchEvent<KubePod>>>,
+    client: KubeClient,
 }
 
-struct Worker {
-    sender: watch::Sender<WatchEvent<KubePod>>,
-    _worker: JoinHandle<()>,
-}
+impl<P: 'static + Provider + Sync + Send> Queue<P> {
+    pub fn new(provider: Arc<P>, client: KubeClient) -> Self {
+        Queue {
+            provider,
+            handlers: HashMap::new(),
+            client,
+        }
+    }
 
-impl Worker {
-    fn create<P>(
+    async fn run_pod(
+        &self,
         initial_event: WatchEvent<KubePod>,
-        provider: Arc<P>,
-        mut error_sender: Sender<(KubePod, anyhow::Error)>,
-    ) -> Self
-    where
-        P: 'static + Provider + Sync + Send,
-    {
-        let (sender, mut receiver) = watch::channel(initial_event);
-        let worker = tokio::spawn(async move {
+    ) -> anyhow::Result<tokio::sync::mpsc::Sender<WatchEvent<KubePod>>> {
+        let (sender, mut receiver) = tokio::sync::mpsc::channel::<WatchEvent<KubePod>>(16);
+
+        let pod_deleted = Arc::new(RwLock::new(false));
```

Comment:
Any reason not to use an `AtomicBool` here?

Reply:
That would work too, seems more heavy than the RwLock though.

Reply:
I honestly don't know which is heavier. Though conceptually in my head it seems like an atomic bool makes more sense

Reply:
The think I like about `RwLock` is that it is async. We don't need any sort of ordering so I tend to think of atomics as unnecessary.